modules/pam_access/pam_access.c

```diff
@@ -620,10 +621,24 @@ from_match (pam_handle_t *pamh UNUSED, char *tok, struct login_info *item)
 	if ((str_len = strlen(string)) > (tok_len = strlen(tok))
 	    && strcasecmp(tok, string + str_len - tok_len) == 0)
 	    return (YES);
-    } else if (item->from_remote_host == 0) {	/* local: no PAM_RHOSTS */
-	if (strcasecmp(tok, "LOCAL") == 0)
-	    return (YES);
-    } else if (tok[(tok_len = strlen(tok)) - 1] == '.') {
+    } else if (strcasecmp(tok, "LOCAL") == 0) {
+	    /* LOCAL matches only local accesses */
+	    if (item->from_remote_host == 0)
+	        return YES;
+	    return NO;
```

Comment:
This `return` is duplicate to the default below.

Reply:
I prefer to keep it.

Reply:
Okay, but in that case
```
return item->from_remote_host ? NO : YES;
```
or at least
```
if (item->from_remote_host)
    return NO;
return YES;
```
would be a bit more readable.  ;-)